### ISSUE TITLE
Scroll view to keep rect into view

### ISSUE THREAD

**Issue by pixtur** [ocornut/imgui#2900]
**Version/Branch of Dear ImGui:**

Version: 1.62
Branch: Master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui.net
Operating System: Win10

**My Issue/Question:**
I'm struggling to scroll an area of a window into view. I studied the documentation and played around with various combinations of `IsRectVisible` and `SetScrollHereY`.

I don't understand, why SetScrollHereY() needs the paramete:. If the out-of-view area was below the visible region, I expect it to be at the bottom. Otherwise at the top. I would prefer a solution like "keep this rectangle visible, but scroll only if necessary and as little as necessary". I know that this is possible, because the Demo-Windows does just that, but can't get it to work.

**Screenshots/Video**

I'm changing an item with cursor-keys up down. The focus is inside a search-field. 
My current (very messy code) looks like this, although I'm not sure if it helps:

```
            if (ImGui.BeginChildFrame(999, new Vector2(200, 200)))
            {
                if (_selectedSymbol == null && SymbolRegistry.Entries.Values.Any())
                    _selectedSymbol = SymbolRegistry.Entries.Values.FirstOrDefault();

                var index = _filter.MatchingSymbols.IndexOf(_selectedSymbol);
                var rectMin = ImGui.GetWindowContentRegionMin() + ImGui.GetWindowPos() + new Vector2(0,(index +1)*ImGui.GetFrameHeight());
                var rectMax = rectMin + new Vector2(10, 10);
                var isVisible = ImGui.IsRectVisible(rectMin, rectMax);

                // Draw debug rect
                ImGui.GetWindowDrawList().AddRectFilled(rectMin, rectMax, isVisible ? Color.Orange : Color.Blue  );

                if (_filterInputType != null)
                {
                    ImGui.PushFont(Fonts.FontSmall);
                    ImGui.TextDisabled(_filterInputType.Name);
                    ImGui.PopFont();
                }

                foreach (var symbol in _filter.MatchingSymbols)
                {
                    ImGui.PushID(symbol.Id.GetHashCode());
                    {
                        ImGui.Selectable("", symbol == _selectedSymbol);

                        if (ImGui.IsItemActivated())
                        {
                            CreateInstance(symbol);
                        }

                        ImGui.SameLine();
                        ImGui.Text(symbol.Name);
                        if (!String.IsNullOrEmpty(symbol.Namespace))
                        {
                            ImGui.TextDisabled(symbol.Namespace);
                            ImGui.SameLine();
                        }
                    }
                    ImGui.PopID();
                }
                if (!isVisible)
                {
                    Log.Debug("Scrolling into view");
                    var keepPos = ImGui.GetCursorScreenPos();
                    ImGui.SetCursorScreenPos(rectMin);
                    ImGui.SetScrollHereY(0);
                    ImGui.SetCursorScreenPos(keepPos);
                    _lastSelectedSymbol = _selectedSymbol;
                }
            }

            ImGui.EndChildFrame();
```

**Comment by ocornut** [comment]
For you can use `ScrollToBringRectIntoView()` in imgui_internal.h do perform what you want.
But it looks like if you are doing "I'm changing an item with cursor-keys up down" you may just as well just enable Keyboard navigation/controls with `io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard` and it'll work.

Please note that 1.62 is rather old at this point so some suggestions/issues may not apply or you may run into different issues than if using latest master.

Linking #2812 there is some work to be done on scrolling helpers/policies.

**Comment by pixtur** [comment]
`ScrollToBringRectIntoView()` would be just the thing I was looking for. Sadly, I can't access it from ImGui.net 🙄 To me it sounds like a method that's worthy to publish in the API.

Regarding `ImGuiConfigFlags_NavEnableKeyboard`: I actually had to disable this flag when the filter input field has focus, because, I want to use keyboard up/down without losing the focus of this field.

I also checked the version I again. Looks like we're at 1.70. I guess should bump up to 1.72.
